sys/KERNELDEFS

```diff
@@ -46,109 +45,69 @@
-endif
-
-ifeq ($(kernel),col)
-MINT = mintv4e.prg
-CPU  = v4e
-KERNELDEFS = -DNO_FAKE_SUPER -DPCI_BIOS
-kernel_nocflags = -DWITH_MMU_SUPPORT
-endif
-
-ifeq ($(kernel),mil)
-MINT = mintmil.prg
-CPU  = 020-60
-KERNELDEFS = -DMILAN -DM68040 -DPCI_BIOS
-endif
-
-ifeq ($(kernel),ara)
-MINT = mintara.prg
-CPU  = 040
-KERNELDEFS = -DM68040 -DWITH_HOSTFS -DBOOTSTRAPABLE -DWITH_NATIVE_FEATURES
-MODULEDIRS = xfs/hostfs
-endif
-
-ifeq ($(kernel),hat)
-MINT = mint000h.prg
-CPU  = 000
-KERNELDEFS = -DM68000 -DOLDTOSFS -DNO_RAMFS -DNO_FAKE_SUPER -DNO_DEV_RANDOM -DWITH_NATIVE_FEATURES
-kernel_nocflags = -DWITH_MMU_SUPPORT
-endif
-
-ifeq ($(kernel),hat030)
-MINT = mint030h.prg
-CPU  = 030
-KERNELDEFS = -DM68030 -DOLDTOSFS -DWITH_NATIVE_FEATURES
-endif
+# for all but m68000
+KERNELNON000DEFS += -DCRYPTO_CODE
+KERNELNON000DEFS += -DSOFT_UNITABLE
+KERNELNON000DEFS += -DBUILTIN_SHELL
+
+
+#
+# These are mapped according to kerneltargets below
+#
+MINT_000 = mint000.prg
+KERNELDEFS_000 = -DM68000 -DNO_RAMFS -DNO_FAKE_SUPER -DNO_DEV_RANDOM
+
+MINT_030 = mint030.prg
+KERNELDEFS_030 = $(KERNELNON000DEFS) -DM68030 $(MMU_SUPPORT)
+
+MINT_040 = mint040.prg
+KERNELDEFS_040 = $(KERNELNON000DEFS) -DM68040 $(MMU_SUPPORT)
+
+MINT_080 = mint080.prg
+KERNELDEFS_080 = $(KERNELNON000DEFS) -DM68040 -DWITH_68080 $(MMU_SUPPORT)
+
+MINT_060 = mint060.prg
+KERNELDEFS_060 = $(KERNELNON000DEFS) -DM68060 -DPCI_BIOS $(MMU_SUPPORT)
+
+MINT_020 = mint020.prg
+KERNELDEFS_020 = $(KERNELNON000DEFS) -DM68000
+
+MINT_deb = mintdeb.prg
+KERNELDEFS_deb = $(KERNELNON000DEFS) -DDEBUG_INFO -DM68040 -DWITH_NATIVE_FEATURES $(MMU_SUPPORT)
+
+MINT_col = mintv4e.prg
+KERNELDEFS_col = $(KERNELNON000DEFS) -DNO_FAKE_SUPER -DPCI_BIOS
+
+MINT_mil = mintmil.prg
+KERNELDEFS_mil = $(KERNELNON000DEFS) -DMILAN -DM68040 -DPCI_BIOS $(MMU_SUPPORT)
+
+MINT_ara = mintara.prg
+KERNELDEFS_ara = $(KERNELNON000DEFS) -DM68040 -DWITH_HOSTFS -DBOOTSTRAPABLE -DWITH_NATIVE_FEATURES $(MMU_SUPPORT)
+MODULEDIRS_ara = xfs/hostfs
+
+MINT_hat = mint000h.prg
+KERNELDEFS_hat = $(KERNELNON000DEFS) -DM68000 -DOLDTOSFS -DNO_RAMFS -DNO_FAKE_SUPER -DNO_DEV_RANDOM -DWITH_NATIVE_FEATURES
+
+MINT_hat030 = mint030h.prg
+KERNELDEFS_hat030 = $(KERNELNON000DEFS) -DM68030 -DOLDTOSFS -DWITH_NATIVE_FEATURES $(MMU_SUPPORT)
 
 # Example for a custom build:
 # Add my1 to "kerneltargets" below
 # Type in your shell "make my1"
-ifeq ($(kernel),my1)
-MINT = mintmy1.prg
-CPU  = 020-60
-KERNELDEFS = -DDEBUG_INFO -DM68040 -DMILAN
-KERNELDEFS += -DMFP_DEBUG_DIRECT
-#KERNELDEFS += -DPROFILING
-#kernel_cflags = -pg
-#kernel_nocflags = -fomit-frame-pointer
-endif
+MINT_my1 = mintmy1.prg
+KERNELDEFS_my1 = -DDEBUG_INFO -DM68040 -DMILAN $(KERNELNON000DEFS) $(MMU_SUPPORT)
+KERNELDEFS_my1 += -DMFP_DEBUG_DIRECT
+#KERNELDEFS_my1 += -DPROFILING
+# add this to ../RULES
+CPU_my1  = 02060
 
 #
 # all default targets
 #
 kerneltargets = 000 020 030 040 060 080 deb mil ara hat hat030 col
-cputargets = 000 020 030 040 060 020-60 v4e
+cputargets = 000 020 030 040 060 02060 v4e
```

Comment:
Since you have been working on this... isn't there a way to get rid of this duplicity? To this day I'm always confused when I should enable/disable entries in `kerneltargets` and `cputargets`. Having just `kerneltargets` would make so much more sense.

Reply:
See above, we need both, because there is no libkernhat.a. Some subdirectories (like usb) also need extra target names like prg, that are not in the toplevel `RULES`